### PR TITLE
Improve wording for the 2 search mode of neural sparse documentation

### PR DESCRIPTION
**Pull request by zhichao-aws**
### Description
The original wording can be misleading since `performance` can be refered as both search relevance and retrieval speed. Here we use search relevance to make it clear. And change the wording to emphasize the intention of both mode.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._

### Version
_List the OpenSearch version to which this PR applies, e.g. 2.14, 2.12--2.14, or all._

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _search-plugins/neural-sparse-search.md [hunk 1]
```diff
@@ -16,8 +16,8 @@ Introduced 2.11
 
 When selecting a model, choose one of the following options:
 
-- Use a sparse encoding model at both ingestion time and search time (high performance, relatively high latency).
-- Use a sparse encoding model at ingestion time and a tokenizer at search time for relatively low performance and low latency. The tokenism doesn't conduct model inference, so you can deploy and invoke a tokenizer using the ML Commons Model API for a more consistent experience.
+- Use a sparse encoding model at both ingestion time and search time for high search relevance with relatively high latency.
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-search.md:19]:
```suggestion
- Use a sparse encoding model at both ingestion time and search time for better search relevance at the expense of relatively high latency.
```

File: _search-plugins/neural-sparse-search.md [hunk 2]
```diff
@@ -16,8 +16,8 @@ Introduced 2.11
...
+- Use a sparse encoding model at ingestion time and a tokenizer at search time for low search latency with relatively low search relevance. The tokenism doesn't conduct model inference, so you can deploy and invoke a tokenizer using the ML Commons Model API for a more consistent experience.
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-search.md:20]:
```suggestion
- Use a sparse encoding model at ingestion time and a tokenizer at search time for lower search latency at the expense of relatively worse search relevance. Tokenization doesn't involve model inference, so you can deploy and invoke a tokenizer using the ML Commons Model API for a more consistent experience.
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @zhichao-aws!

File: _search-plugins/neural-sparse-search.md (hunk 2 above)
Comment (natebower) [reply _search-plugins/neural-sparse-search.md:20]: Changed "worse" to "lower" with regard to "relevance".

**Review by natebower — APPROVED** [review]
LGTM with noted suggestions.

**Pull request merged**